Re-optimize and clean up settings, social and logo icons

pixelipo opened the pull request:
I've discovered a whole set of icons in settings app that needed to be fixed (mostly I just replaced them with already optimized icons found in **core/img/*** folders).

I've also fixed social icons as well as logo and favicon SVGs. Those last ones are quite critical, so test those carefully.

@nextcloud/designers 

(finally) Fixes #4076

danxuliu committed:
Add proper handling of files without permissions

Now a file gets its directory permissions only if it contained no
permissions (they were undefined or null), but not if its permissions
were set to "NONE".

Besides that, now file actions that do not require any permission on the
file to be performed can be used on files that have no permissions.

Signed-off-by: Daniel Calviño Sánchez <danxuliu@gmail.com>

danxuliu committed:
Remove requeriment for read permission on some actions

The Details and the Favorite actions do not require any permission on
the files to be performed.

Signed-off-by: Daniel Calviño Sánchez <danxuliu@gmail.com>

danxuliu committed:
Fix asserts silently not executed

The first parameter of "apply" must be the object to act as "this", and
the Promise callback gets the parameters provided in the "resolve".

Signed-off-by: Daniel Calviño Sánchez <danxuliu@gmail.com>

danxuliu committed:
Fix tests for parsing of permissions

Now that these tests are executed they are revealed to be partially
obsolete; they were fixed to match the current parsing behaviour.

Signed-off-by: Daniel Calviño Sánchez <danxuliu@gmail.com>

danxuliu committed:
Set read permission for files based on the data returned by the server

Now that the permissions returned by the server specify whether a file
is readable or not the frontend no longer needs to assume that every
file is readable.

Signed-off-by: Daniel Calviño Sánchez <danxuliu@gmail.com>

danxuliu committed:
Return whether the file is readable or not in the DAV permissions

Until now it was safe to assume that every file was readable by its
owner, so there was no need to return whether the file was readable or
not. However, with the introduction of end to end encryption that is no
longer the case, and it is now necessary to explicitly provide that
information.

Signed-off-by: Daniel Calviño Sánchez <danxuliu@gmail.com>

rullzer committed:
If there is no internet connection don't try to contact appstore

Fixes #7119

Signed-off-by: Roeland Jago Douma <roeland@famdouma.nl>

danxuliu committed:
Force breaks on extremely long words

If not broken extremely long words overflow their container.

Signed-off-by: Daniel Calviño Sánchez <danxuliu@gmail.com>

skjnldsv committed:
Add app content list

Signed-off-by: John Molakvoæ (skjnldsv) <skjnldsv@protonmail.com>

skjnldsv committed:
Checkboxes and various design fixes

Signed-off-by: John Molakvoæ (skjnldsv) <skjnldsv@protonmail.com>

skjnldsv committed:
Checkboxes and favourite fix

Signed-off-by: John Molakvoæ (skjnldsv) <skjnldsv@protonmail.com>

pixelipo committed:
Re-optimize and clean up settings, social and logo icons

Signed-off-by: Marin Treselj <marin@pixelipo.com>

pixelipo committed:
Re-optimize and clean up settings, social and logo icons

Signed-off-by: Marin Treselj <marin@pixelipo.com>

blizzz committed:
only replace permission popupmenu

the contacts popovermenu is also present and is being replaces, ending
up in two permission popupmenus with checkboxes duplicating the id,
breaking further permission changes.

plus, fixing a selector

Signed-off-by: Arthur Schiwon <blizzz@arthur-schiwon.de>

jancborchardt committed:
Merge pull request #7163 from nextcloud/force-breaks-on-extremely-long-words-in-comments

Force breaks on extremely long words in comments

rullzer committed:
Fix tests

Signed-off-by: Roeland Jago Douma <roeland@famdouma.nl>

rullzer committed:
Merge pull request #7121 from nextcloud/fix_7119

If there is no internet connection don't try to contact appstore

nextcloud-bot committed:
[tx-robot] updated from transifex

MorrisJobke reviewed (approved):
Tested and works 👍

skjnldsv commented on settings/img/users.svg:
```diff
@@ -1 +1 @@
-<svg xmlns="http://www.w3.org/2000/svg" height="16" width="16" version="1"><path color="#000" fill="none" d="M-62.897-32.993h163.31v97.986h-163.31z"/><path style="block-progression:tb;text-transform:none;text-indent:0" d="M4.118 1.71c-1.113 0-2.058.815-2.058 1.866.008.332.038.742.236 1.608v.02l.022.022c.063.183.156.287.278.43.123.14.27.31.407.45l.043.042c.028.12.06.25.086.365.066.307.06.525.043.6 ...
\ No newline at end of file
+<svg xmlns="http://www.w3.org/2000/svg" viewbox="0 0 16 16" height="16" width="16" version="1.1"><path d="m9 1c-1.746 0-3 1.43-3 2.8 0 1.4 0.1 2.4 0.8 3.5 0.224 0.29 0.485 0.35 0.7 0.6 0.135 0.5 0.24 1 0.1 1.5-0.436 0.153-0.85 0.332-1.27 0.5-0.51-0.273-1.1-0.5-1.61-0.7-0.07-0.28-0.02-0.487 0.05-0.75 0.12-0.125 0.23-0.18 0.36-0.3 0.37-0.45 0.39-1.21 0.39-1.75 0-0.8-0.72-1.4-1.5-1.4-0.87 0-1.5 0.72 ...
```
Not convinced with that one. Too small. I prefer the old one :)

jancborchardt replied:
Yeah, this shouldn’t be changed in size. The form is visually not filled enough. :) Can you fix this @pixelipo, then it’s good to go.

nextcloud-bot committed:
[tx-robot] updated from transifex

nextcloud-bot committed:
[tx-robot] updated from transifex

MorrisJobke committed:
Merge pull request #7211 from nextcloud/fix-5916

only replace permission popupmenu

jancborchardt committed:
Reduce size of date in list

Signed-off-by: Jan-Christoph Borchardt <hey@jancborchardt.net>

jancborchardt committed:
Move star to top right instead of top left, fix checkbox overlay

Signed-off-by: Jan-Christoph Borchardt <hey@jancborchardt.net>

jancborchardt committed:
Increase clickable area of actions to 44*44px

Signed-off-by: Jan-Christoph Borchardt <hey@jancborchardt.net>

MorrisJobke committed:
Merge pull request #7195 from nextcloud/app-content-list

Add app content list

schiessle committed:
Merge pull request #7047 from nextcloud/add-support-for-files-with-no-permissions

Add support for files with no permissions

pixelipo committed:
Make user-related icons bigger.

Signed-off-by: Marin Treselj <marin@pixelipo.com>

pixelipo committed:
Make user-related icons bigger.

Signed-off-by: Marin Treselj <marin@pixelipo.com>

pixelipo commented:
Ok @skjnldsv @jancborchardt  - now that icon fills more space. I didn't revert the old icon since it wasn't using the same "figure" as the other "personal" icons.

I've also updated all the other user-related icons to the same "figure" shape:
```	modified:   core/img/actions/user.svg
	modified:   core/img/categories/social.svg
	modified:   core/img/filetypes/text-vcard.svg
	modified:   settings/img/personal.png
	modified:   settings/img/personal.svg
	modified:   settings/img/users.svg
```

Ready to review and merge :)

MorrisJobke commented:
> Ready to review and merge :)

There are conflicts :/

pixelipo committed:
Merge branch 'fix-broken-svg-4' of github.com:nextcloud/server into fix-broken-svg-4

pixelipo commented:
Screwed up with rebase, please go here: #7231
